api/t8_mesh_handle/competences.hxx

```diff
@@ -0,0 +1,99 @@
+/*
+This file is part of t8code.
+t8code is a C library to manage a collection (a forest) of multiple
+connected adaptive space-trees of general element classes in parallel.
+
+Copyright (C) 2025 the developers
+
+t8code is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+t8code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with t8code; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file competences.hxx
+ * Definition of the additional competences/functionalities that can be used for the mesh class.
+ * Especially, competences to cache functionalities of elements instead of calculating them each time a function
+ * is called are provided.
+ *
+ * All competences have the same inheritance pattern: 
+ * We use the CRTP pattern as we may need to access members of the derived class \ref t8_mesh_handle::element. 
+ * The t8_crtp_operator is used for convenience/clear code (avoid to type a static cast explicitly each time 
+ * we need functionality of TUnderlying).
+ * Especially for the competences to cache functionality, the access of members is not necessary, 
+ * such that it is not obvious why we use the crtp. For competences that extend the functionality of the element, 
+ * this is required. 
+ * We use it for all competences for consistency and compatibility with the \ref t8_mesh_handle::element class.
+ */
+
+#ifndef T8_COMPETENCES_HXX
+#define T8_COMPETENCES_HXX
+
+#include <t8.h>
+#include <t8_types/t8_operators.hxx>
+#include <t8_types/t8_vec.hxx>
+#include <t8_eclass.h>
+#include <array>
+#include <vector>
+#include <optional>
+
+namespace t8_mesh_handle
+{
+
+/**
+ * Competence to cache the vertex coordinates of an element at the first function call.
+ * \tparam TUnderlying Use the \ref element with specified competences as template parameter.
+ */
+template <typename TUnderlying>
+struct cache_vertex_coordinates: t8_crtp_operator<TUnderlying, cache_vertex_coordinates>
```

Comment:
The crtp operator is a little overkill and in some scenarios it can mess with inheritance. Since we do not need the additional features it implements, you can better use [`t8_crtp_basic`](https://github.com/DLR-AMR/t8code/blob/11241199d92ed4ca2427377f06aef5b56c4665eb/src/t8_types/t8_crtp.hxx#L41):
```suggestion
struct cache_vertex_coordinates: t8_crtp_basic<TUnderlying>
```

Reply:
Oh yes, cool implementation. Btw we dont need the iostream include in the crtp file, can i just remove it in this PR?